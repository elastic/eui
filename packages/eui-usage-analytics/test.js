/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const Codeowners = require('codeowners');
const temp = new Codeowners("../../../kibana");

const { scan } = require('./scan');

const runScan = async () => {
  const scanResult = await scan();
  console.log(scanResult);
};

runScan();
