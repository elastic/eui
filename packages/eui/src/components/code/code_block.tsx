/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, useMemo } from 'react';
import { RefractorNode } from 'refractor';
import classNames from 'classnames';

import {
  useCombinedRefs,
  useEuiTheme,
  useEuiMemoizedStyles,
} from '../../services';
import { ExclusiveUnion } from '../common';
import {
  EuiCodeSharedProps,
  DEFAULT_LANGUAGE,
  checkSupportedLanguage,
  getHtmlContent,
  highlightByLine,
} from './utils';
import { LineAnnotationMap } from './code_block_annotations';
import { useOverflow } from './code_block_overflow';
import { useCopy } from './code_block_copy';
import {
  useFullScreen,
  EuiCodeBlockFullScreenWrapper,
} from './code_block_full_screen';
import { EuiCodeBlockControls } from './code_block_controls';
import { EuiCodeBlockVirtualized } from './code_block_virtualized';
import {
  euiCodeBlockStyles,
  euiCodeBlockPreStyles,
  euiCodeBlockCodeStyles,
} from './code_block.styles';

// Based on observed line height for non-virtualized code blocks
const fontSizeToRowHeightMap = {
  s: 18,
  m: 21,
  l: 24,
};

export const FONT_SIZES = ['s', 'm', 'l'] as const;
export type EuiCodeBlockFontSize = (typeof FONT_SIZES)[number];

export const PADDING_SIZES = ['none', 's', 'm', 'l'] as const;
export type EuiCodeBlockPaddingSize = (typeof PADDING_SIZES)[number];

// This exclusive union enforces specific props based on isVirtualized
type VirtualizedOptionProps = ExclusiveUnion<
  {
    isVirtualized: true;
    overflowHeight: number | string;
    whiteSpace?: 'pre';
  },
  {
    isVirtualized?: false;
    overflowHeight?: number | string;
    whiteSpace?: 'pre' | 'pre-wrap';
  }
>;

interface LineNumbersConfig {
  start?: number;
  highlight?: string;
  annotations?: LineAnnotationMap;
}

export type EuiCodeBlockProps = EuiCodeSharedProps & {
  paddingSize?: EuiCodeBlockPaddingSize;
  fontSize?: EuiCodeBlockFontSize;

  /**
   * Specify how `white-space` inside the element is handled.
   * `pre` respects line breaks/white space but doesn't force them to wrap the line
   * `pre-wrap` respects line breaks/white space but does force them to wrap the line when necessary.
   */
  whiteSpace?: 'pre' | 'pre-wrap';

  /**
   * Displays an icon button to copy the code snippet to the clipboard.
   */
  isCopyable?: boolean;

  /**
   * Displays line numbers.
   * Optionally accepts a configuration object for setting the starting number,
   * visually highlighting ranges, or annotating specific lines:
   * `{ start: 100, highlight: '1, 5-10, 20-30, 40', annotations: { 6: 'A special note about this line' } }`
   */
  lineNumbers?: boolean | LineNumbersConfig;

  /**
   * Sets the maximum container height.
   * Accepts a pixel value (`300`) or a percentage (`'100%'`)
   * Ensure the container has calcuable height when using a percentage
   */
  overflowHeight?: number | string;

  /**
   * Renders code block lines virtually.
   * Useful for improving load times of large code blocks.
   *
   * When using this configuration, `overflowHeight` is required and
   * `whiteSpace` can only be `pre`.
   */
  isVirtualized?: boolean;
} & VirtualizedOptionProps;

export const EuiCodeBlock: FunctionComponent<EuiCodeBlockProps> = ({
  language: _language = DEFAULT_LANGUAGE,
  transparentBackground = false,
  paddingSize = 'l',
  fontSize = 's',
  isCopyable = false,
  whiteSpace = 'pre-wrap',
  children,
  className,
  overflowHeight,
  isVirtualized: _isVirtualized,
  lineNumbers = false,
  ...rest
}) => {
  const euiTheme = useEuiTheme();
  const language = useMemo(
    () => checkSupportedLanguage(_language),
    [_language]
  );

  const lineNumbersConfig = useMemo(() => {
    const config = typeof lineNumbers === 'object' ? lineNumbers : {};
    return lineNumbers
      ? { start: 1, show: true, ...config }
      : { start: 1, show: false };
  }, [lineNumbers]);

  // Used by `FixedSizeList` when `isVirtualized=true` or `children` is parsable
  const data: RefractorNode[] = useMemo(() => {
    if (typeof children !== 'string') {
      return [];
    }
    return highlightByLine(children, language, lineNumbersConfig, euiTheme);
  }, [children, language, lineNumbersConfig, euiTheme]);

  // Used by `pre` when `isVirtualized=false` or `children` is not parsable
  const content = useMemo(
    () => getHtmlContent(data, children),
    [data, children]
  );

  const isVirtualized = useMemo(
    () => !!(_isVirtualized && Array.isArray(data)),
    [_isVirtualized, data]
  );

  const { innerTextRef, copyButton } = useCopy({
    isCopyable,
    isVirtualized,
    children,
  });

  const { setWrapperRef, tabIndex, overflowHeightStyles } = useOverflow({
    overflowHeight,
  });

  const combinedRef = useCombinedRefs<HTMLPreElement>([
    innerTextRef,
    setWrapperRef,
  ]);

  const { fullScreenButton, isFullScreen, onKeyDown } = useFullScreen({
    overflowHeight,
  });

  const hasControls = !!(copyButton || fullScreenButton);
  const hasBothControls = !!(copyButton && fullScreenButton);

  const styles = useEuiMemoizedStyles(euiCodeBlockStyles);
  const cssStyles = [
    styles.euiCodeBlock,
    styles[fontSize],
    transparentBackground && styles.transparentBackground,
    hasControls &&
      (hasBothControls
        ? styles.hasBothControls[paddingSize]
        : styles.hasControls[paddingSize]),
  ];

  const preStyles = useEuiMemoizedStyles(euiCodeBlockPreStyles);
  const [preProps, preFullscreenProps] = useMemo(() => {
    const isWhiteSpacePre = whiteSpace === 'pre' || isVirtualized;

    const cssStyles = [
      preStyles.euiCodeBlock__pre,
      isWhiteSpacePre
        ? preStyles.whiteSpace.pre.pre
        : preStyles.whiteSpace.preWrap.preWrap,
    ];

    const preProps = {
      className: 'euiCodeBlock__pre',
      css: [
        ...cssStyles,
        preStyles.padding[paddingSize],
        hasControls &&
          (isWhiteSpacePre
            ? preStyles.whiteSpace.pre.controlsOffset[paddingSize]
            : preStyles.whiteSpace.preWrap.controlsOffset[paddingSize]),
      ],
      tabIndex: isVirtualized ? 0 : tabIndex,
    };
    const preFullscreenProps = {
      className: 'euiCodeBlock__pre',
      css: [
        ...cssStyles,
        // Force fullscreen to use xl padding
        preStyles.padding.xl,
        hasControls &&
          (isWhiteSpacePre
            ? preStyles.whiteSpace.pre.controlsOffset.xl
            : preStyles.whiteSpace.preWrap.controlsOffset.xl),
      ],
      tabIndex: 0,
      onKeyDown,
    };

    return [preProps, preFullscreenProps];
  }, [
    preStyles,
    whiteSpace,
    isVirtualized,
    hasControls,
    paddingSize,
    onKeyDown,
    tabIndex,
  ]);

  const codeStyles = useEuiMemoizedStyles(euiCodeBlockCodeStyles);
  const codeProps = useMemo(() => {
    const cssStyles = [
      codeStyles.euiCodeBlock__code,
      isVirtualized && codeStyles.isVirtualized,
    ];

    return {
      className: 'euiCodeBlock__code',
      css: cssStyles,
      'data-code-language': language,
      ...rest,
    };
  }, [codeStyles, language, isVirtualized, rest]);

  return (
    <div
      css={cssStyles}
      className={classNames('euiCodeBlock', className)}
      style={overflowHeightStyles}
    >
      {isVirtualized ? (
        <EuiCodeBlockVirtualized
          data={data}
          rowHeight={fontSizeToRowHeightMap[fontSize]}
          overflowHeight={overflowHeight}
          preProps={preProps}
          codeProps={codeProps}
        />
      ) : (
        <pre {...preProps} ref={combinedRef} style={overflowHeightStyles}>
          <code {...codeProps}>{content}</code>
        </pre>
      )}
      <EuiCodeBlockControls
        controls={[fullScreenButton, copyButton]}
        paddingSize={paddingSize}
      />

      {isFullScreen && (
        <EuiCodeBlockFullScreenWrapper>
          {isVirtualized ? (
            <EuiCodeBlockVirtualized
              data={data}
              rowHeight={fontSizeToRowHeightMap.l}
              preProps={preFullscreenProps}
              codeProps={codeProps}
            />
          ) : (
            <pre {...preFullscreenProps}>
              <code {...codeProps}>{content}</code>
            </pre>
          )}
          <EuiCodeBlockControls
            controls={[fullScreenButton, copyButton]}
            paddingSize="l"
          />
        </EuiCodeBlockFullScreenWrapper>
      )}
    </div>
  );
};
