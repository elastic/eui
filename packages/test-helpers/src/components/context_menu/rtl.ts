/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { fireEvent, waitFor } from '@testing-library/react';

const waitForPanelTransition = async () => {
  // Used document instead of container or screen due to context menus living in portals
  const getPanels = () => document.querySelectorAll('.euiContextMenuPanel');

  // 2 panels will appear for the transition animation
  await waitFor(() => {
    expect(getPanels().length).toEqual(2);
  });

  // Outgoing panel will be removed on animation end
  fireEvent.animationEnd(getPanels()[0]);
  if (getPanels().length > 1) {
    fireEvent.animationEnd(getPanels()[1]);
  }

  // Transition/animation is done once we're back to 1 panel
  expect(getPanels().length).toEqual(1);
};

export const EuiContextMenuTestHelpers = {
  waitForPanelTransition,
};
