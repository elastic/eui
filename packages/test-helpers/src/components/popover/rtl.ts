/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { waitFor } from '@testing-library/react';

const waitForOpen = () => {
  return waitFor(() => {
    const openPopover = document.querySelector('[data-popover-open]');
    expect(openPopover).toBeTruthy();
  });
}

const waitForClose = () => {
  return waitFor(() => {
    const openPopover = document.querySelector('[data-popover-open]');
    expect(openPopover).toBeFalsy();
  });
};

export const EuiPopoverTestHelpers = {
  waitForOpen,
  waitForClose,
};
