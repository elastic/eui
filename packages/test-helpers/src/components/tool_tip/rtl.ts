/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { waitFor } from '@testing-library/react';

const waitForVisible = () => {
  return waitFor(
    () => {
      const tooltip = document.querySelector('.euiToolTipPopover');
      expect(tooltip).toBeVisible();
    },
    { timeout: 3000 } // Account for long delay on tooltips
  );
};

const waitForHidden = () => {
  return waitFor(() => {
    const tooltip = document.querySelector('.euiToolTipPopover');
    expect(tooltip).toBeNull();
  });
};

export const EuiToolTipTestHelpers = {
  waitForVisible,
  waitForHidden,
};
