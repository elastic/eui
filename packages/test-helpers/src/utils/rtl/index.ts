/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

export { dataTestSubjQueries } from './data_test_subj_queries';
export { withEuiProvider } from './with_eui_provider'
export { withQueries } from './with_queries';
