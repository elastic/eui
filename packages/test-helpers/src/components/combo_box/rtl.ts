/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { fireEvent, waitFor } from '@testing-library/react';
import { screen } from '@testing-library/react';
import { EuiPopoverTestHelpers } from '../popover/rtl';

const showOptions = async () => {
  fireEvent.click(screen.getByTestSubject('comboBoxToggleListButton'));
  await EuiPopoverTestHelpers.waitForOpen();
  await waitFor(() => {
    expect(screen.getByRole('listbox')).toBeInTheDocument();
  });
};

export const EuiComboBoxTestHelpers = {
  showOptions,
};
