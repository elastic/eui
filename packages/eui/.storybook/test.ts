/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { queries, within, waitFor, fireEvent, expect } from '@storybook/test';
import * as dataTestSubjQueries from '../src/test/rtl/data_test_subj_queries';

/**
 * Custom Storybook within util with EUI query helpers
 * + additional chained async/waitFor component utils
 *
 * @see https://storybook.js.org/docs/writing-stories/play-function#writing-stories-with-the-play-function
 * @see https://testing-library.com/docs/dom-testing-library/api-within/
 */
const customWithin = (canvasElement: HTMLElement) => {
  const canvas = within<typeof queries & typeof dataTestSubjQueries>(
    canvasElement,
    { ...queries, ...dataTestSubjQueries }
  );

  return {
    ...canvas,

    /**
     * 1. Loki doesn't like userEvent, only fireEvent
     * 2. Storybook fires fireEvents too early (esp. on page load), so we add a waitFor
     */
    waitForAndClick: async (testSubject: string) => {
      await waitFor(() =>
        expect(canvas.getByTestSubject(testSubject)).toBeInTheDocument()
      );
      await fireEvent.click(canvas.getByTestSubject(testSubject));
    },

    waitForEuiPopoverVisible: async () =>
      await waitFor(() =>
        expect(canvasElement.querySelector('[data-popover-open]')).toBeVisible()
      ),
    waitForEuiPopoverHidden: async () =>
      await waitFor(() =>
        expect(
          canvasElement.querySelector('[data-popover-panel]')
        ).not.toBeInTheDocument()
      ),
  };
};

export { customWithin as within };
