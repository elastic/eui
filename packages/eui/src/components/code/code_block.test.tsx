/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import { requiredProps } from '../../test/required_props';
import { render } from '../../test/rtl';

import { EuiCodeBlock, FONT_SIZES, PADDING_SIZES } from './code_block';

const code = `var some = 'code';
console.log(some);`;

describe('EuiCodeBlock', () => {
  it('renders a code block', () => {
    const { container } = render(
      <EuiCodeBlock {...requiredProps}>{code}</EuiCodeBlock>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('props', () => {
    describe('transparentBackground', () => {
      it('is rendered', () => {
        const { container } = render(
          <EuiCodeBlock transparentBackground>{code}</EuiCodeBlock>
        );

        expect(container.firstChild).toMatchSnapshot();
      });
    });

    describe('isCopyable', () => {
      it('is rendered', () => {
        const { container } = render(
          <EuiCodeBlock isCopyable>{code}</EuiCodeBlock>
        );

        expect(container.firstChild).toMatchSnapshot();
      });
    });

    describe('overflowHeight', () => {
      it('is rendered', () => {
        const { container } = render(
          <EuiCodeBlock overflowHeight={200}>{code}</EuiCodeBlock>
        );

        expect(container.firstChild).toMatchSnapshot();
      });
    });

    describe('language', () => {
      it('is rendered', () => {
        const { container } = render(
          <EuiCodeBlock language="html">{code}</EuiCodeBlock>
        );

        expect(container.firstChild).toMatchSnapshot();
      });
    });

    describe('fontSize', () => {
      FONT_SIZES.forEach((fontSize) => {
        test(`${fontSize} is rendered`, () => {
          const { container } = render(
            <EuiCodeBlock fontSize={fontSize}>{code}</EuiCodeBlock>
          );

          expect(container.firstChild).toMatchSnapshot();
        });
      });
    });

    describe('paddingSize', () => {
      PADDING_SIZES.forEach((paddingSize) => {
        test(`${paddingSize} is rendered`, () => {
          const { container } = render(
            <EuiCodeBlock paddingSize={paddingSize}>{code}</EuiCodeBlock>
          );

          expect(container.firstChild).toMatchSnapshot();
        });
      });
    });

    describe('whiteSpace', () => {
      it('renders a pre block tag with a css class modifier', () => {
        const { container } = render(
          <EuiCodeBlock whiteSpace="pre" {...requiredProps}>
            {code}
          </EuiCodeBlock>
        );
        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });

  describe('dynamic content', () => {
    it('updates DOM when input changes', () => {
      const { container, rerender } = render(
        <EuiCodeBlock language="javascript">
          const value = &apos;State 1&apos;
        </EuiCodeBlock>
      );

      expect(container).toMatchSnapshot();

      rerender(
        <EuiCodeBlock language="javascript">
          const value = &apos;State 2&apos;
        </EuiCodeBlock>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('fullscreen', () => {
    it('displays content in fullscreen mode', () => {
      const component = mount(
        <EuiCodeBlock
          {...requiredProps}
          language="javascript"
          overflowHeight={300}
        >
          const value = &quot;hello&quot;
        </EuiCodeBlock>
      );
      component.find('button[aria-label="Expand"]').simulate('click');
      component.update();
      expect(
        component.find('div.euiCodeBlockFullScreen').render()
      ).toMatchSnapshot();
    });

    it('closes fullscreen mode when the escape key is pressed', () => {
      const component = mount(
        <EuiCodeBlock
          {...requiredProps}
          language="javascript"
          overflowHeight={300}
        >
          const value = &quot;world&quot;
        </EuiCodeBlock>
      );
      component.find('button[aria-label="Expand"]').simulate('click');
      component.update();
      component
        .find('div.euiCodeBlockFullScreen')
        .find('pre.euiCodeBlock__pre')
        .simulate('keyDown', { key: 'Escape' });
      expect(component.find('.euiCodeBlockFullScreen')).toHaveLength(0);
    });
  });

  describe('virtualization', () => {
    it('renders a virtualized code block', () => {
      const { container } = render(
        <EuiCodeBlock
          isVirtualized={true}
          overflowHeight="50%"
          {...requiredProps}
        >
          {code}
        </EuiCodeBlock>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    describe('type checks', () => {
      it('requires overflowHeight', () => {
        // @ts-expect-error should expect overflowHeight
        shallow(<EuiCodeBlock isVirtualized overflowHeight={undefined} />);
      });

      it('only allows whiteSpace of pre', () => {
        shallow(
          // @ts-expect-error should only accept "pre"
          <EuiCodeBlock
            isVirtualized
            overflowHeight={50}
            whiteSpace="pre-wrap"
          />
        );
        // OK
        shallow(
          <EuiCodeBlock isVirtualized overflowHeight={50} whiteSpace="pre" />
        );
      });
    });
  });

  describe('line numbers', () => {
    it('renders line numbers', () => {
      const { container } = render(
        <EuiCodeBlock lineNumbers {...requiredProps}>
          {code}
        </EuiCodeBlock>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders line numbers with a start value', () => {
      const { container } = render(
        <EuiCodeBlock lineNumbers={{ start: 10 }} {...requiredProps}>
          {code}
        </EuiCodeBlock>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders highlighted line numbers', () => {
      const { container } = render(
        <EuiCodeBlock lineNumbers={{ highlight: '1' }} {...requiredProps}>
          {code}
        </EuiCodeBlock>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders annotated line numbers', () => {
      const { container } = render(
        <EuiCodeBlock
          lineNumbers={{ annotations: { 1: 'hello world' } }}
          {...requiredProps}
        >
          {code}
        </EuiCodeBlock>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
