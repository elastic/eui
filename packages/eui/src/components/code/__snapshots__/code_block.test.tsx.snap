// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiCodeBlock Virtualization renders a virtualized code block 1`] = `
<div
  class="euiCodeBlock testClass1 testClass2 emotion-euiCodeBlock-s-hasControls-euiTestCss"
  style="block-size: 50%;"
>
  <div
    data-eui="EuiAutoSizer"
  >
    <pre
      class="euiCodeBlock__pre emotion-euiCodeBlock__pre-pre-padding-controlsOffset"
      style="position: relative; block-size: 600px; inline-size: 600px; overflow: auto; will-change: transform; direction: ltr;"
      tabindex="0"
    >
      <code
        aria-label="aria-label"
        class="euiCodeBlock__code emotion-euiCodeBlock__code-isVirtualized"
        data-code-language="text"
        data-test-subj="test subject string"
        style="block-size: 36px; inline-size: 100%;"
      >
        <span
          class="euiCodeBlock__line emotion-euiCodeBlock__line"
          style="position: absolute; inset-inline-start: 0; inset-block-start: 0; block-size: 18px; inline-size: 100%;"
        >
          var some = 'code';

        </span>
        <span
          class="euiCodeBlock__line emotion-euiCodeBlock__line"
          style="position: absolute; inset-inline-start: 0; inset-block-start: 18px; block-size: 18px; inline-size: 100%;"
        >
          console.log(some);
        </span>
      </code>
    </pre>
  </div>
  <div
    class="euiCodeBlock__controls emotion-euiCodeBlock__controls-l"
  >
    <button
      aria-label="Expand"
      class="euiButtonIcon euiCodeBlock__fullScreenButton emotion-euiButtonIcon-xs-empty-text"
      type="button"
    >
      <span
        aria-hidden="true"
        class="euiButtonIcon__icon"
        color="inherit"
        data-euiicon-type="fullScreen"
      />
    </button>
  </div>
</div>
`;

exports[`EuiCodeBlock renders a code block 1`] = `
<div
  class="euiCodeBlock testClass1 testClass2 emotion-euiCodeBlock-s-euiTestCss"
>
  <pre
    class="euiCodeBlock__pre emotion-euiCodeBlock__pre-preWrap-padding"
    tabindex="-1"
  >
    <span
      aria-hidden="true"
      class="euiScreenReaderOnly"
      data-tabular-copy-marker="no-copy"
    >
      ✄𐘗
    </span>
    <div
      class="emotion-euiScreenReaderOnly"
    >
      text code block:
    </div>
    <span
      aria-hidden="true"
      class="euiScreenReaderOnly"
      data-tabular-copy-marker="no-copy"
    >
      ✄𐘗
    </span>
    <code
      aria-label="aria-label"
      class="euiCodeBlock__code emotion-euiCodeBlock__code"
      data-code-language="text"
      data-test-subj="test subject string"
    >
      <span
        class="euiCodeBlock__line emotion-euiCodeBlock__line"
      >
        var some = 'code';

      </span>
      <span
        class="euiCodeBlock__line emotion-euiCodeBlock__line"
      >
        console.log(some);
      </span>
    </code>
  </pre>
</div>
`;
